### DISCUSSION TITLE
Welcome to WikidPad Discussions!

### DISCUSSION THREAD

**Discussion by gchriz** [WikidPad/WikidPad#71] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by gchriz** [comment]
Hi!  I just activated the new (beta) feature "Discussions" here.

Don't know whether it's a good idea... We will see.

**Comment by BjornFJohansson** [comment]
This could be a good idea.

**Comment by spicedreams** [comment]
Views tree: usefulness of alias, font, icon and so on, and how they can be hidden

I am increasingly using the Views tree to organise my tasks in a form of Getting Things Done.  One aspect is what the author calls contexts, such as places- so you have a way to generate a list of things to do when you are in that place. I use the notation
[context: here; there; everywhere] thing to do
which creates (if not there already) a subtree named context with branches here, there and everywhere and pointers from each of those branches to the page containing the notation. 
That works fine (and I have a few other tree-structured classifications like that too). 
But in that same Views list are items alias, font, icon, pagetype, template which are no use to me (in the Views tree).

Is there a simple way to remove them from the Views tree? Is there a negative consequence to doing that?

**Reply by vashnt707** [reply]
Code to add/remove items to View node is here : https://github.com/WikidPad/WikidPad/blob/master/WikidPad/lib/pwiki/WikiTreeCtrl.py#L571 

You can disable/delete them as needed. For instance, you can create a new settings for allowed/blocked attributes, put "context" in there and filter them within AttrCategoryNode at L574

**Reply by spicedreams** [reply]
Thanks you for the very prompt response. I will have to explore how that code works.
Would I make a copy of WikiTreeCtrl.py in my user_extensions directory, so that my changes might survive (some) version upgrades?

**Reply by spicedreams** [reply]
I tried making a copy in my user_extensions directory, in which I commented out the lines that add the "undefined nodes" subtree. After saving, then restarting WikidPad, my changes were ignored, so that approach probably doesn't work.

**Reply by vashnt707** [reply]
Are you sure that file got loaded? Not seeing any errors in log message? I see relative imports in that file and not sure how that would work if it was copied to extensions dir (I am assuming the files are executed under eval/exec or something and hence shouldn't be an issue)

Or the WikiTreeCtrl is initialized before the plugins get loaded and hence new changes do not reflect. I would request someone else to put their view on this. I run from source, hence I directly edit and keep track of my changes.

**Reply by spicedreams** [reply]
Actually there is an error in the log (not that I can interpret it)

Version: 'wikidPad 2.3beta16' Session start: 2021-02-25 17:10:52
Traceback (most recent call last):
  File "lib/pwiki/PluginManager.py", line 341, in loadPlugins
    mbcsEnc(fullname)[0], (".py", "r", imp.PY_SOURCE))
  File "/home/graham/bin/WikidPad2_3beta16/user_extensions/WikidMap.py", line 17, in <module>
    import wx.html2
ImportError: No module named html2
Traceback (most recent call last):
  File "lib/pwiki/PluginManager.py", line 341, in loadPlugins
    mbcsEnc(fullname)[0], (".py", "r", imp.PY_SOURCE))
  File "/home/graham/bin/WikidPad2_3beta16/user_extensions/WikiTreeCtrl.py", line 10, in <module>
    import customtreectrl
ImportError: No module named customtreectrl
Traceback (most recent call last):
  File "lib/pwiki/PluginManager.py", line 341, in loadPlugins
    mbcsEnc(fullname)[0], (".py", "r", imp.PY_SOURCE))
  File "/home/graham/bin/WikidPad2_3beta16/user_extensions/WikidMap.py", line 17, in <module>
    import wx.html2
ImportError: No module named html2
Traceback (most recent call last):
  File "lib/pwiki/PluginManager.py", line 341, in loadPlugins
    mbcsEnc(fullname)[0], (".py", "r", imp.PY_SOURCE))
  File "/home/graham/bin/WikidPad2_3beta16/user_extensions/WikiTreeCtrl.py", line 10, in <module>
    import customtreectrl
ImportError: No module named customtreectrl

**Reply by vashnt707** [reply]
Yeah as mentioned above, this file has relative imports and thus breaking. For instance, at https://github.com/WikidPad/WikidPad/blob/master/WikidPad/lib/pwiki/WikiTreeCtrl.py#L10, instead of `from .`, it could be relative to package root such as `from lib.pwiki` in which case if you put this modified file in user_directory, it would pick up the `customtreectrl` dependency correctly.

Not tough but not straightforward if you are new to this

**Comment by spicedreams** [comment]
That solved the problem of loading customtreectrl but the "undefined nodes" tree entry remains. 
Thanks for your help, we have both spent enough effort on my essentially cosmetic problem....

**Comment by ghost** [comment]
Hi all!
